### ISSUE TITLE
Cloud API adaptor on Azure unbound variable

### ISSUE THREAD

**Issue by cmaf** [confidential-containers/cloud-api-adaptor#1682]
Following the steps for setting up CAA on Azure, https://confidentialcontainers.org/docs/cloud-api-adaptor/azure/, after running `CLOUD_PROVIDER=azure make deploy` the operator install daemon pod fails and the logs show an unbound variable:

```
$ kubectl describe pod cc-operator-daemon-install-tr6fx -n confidential-containers-system
  Events:
  Type     Reason     Age                    From               Message
  ----     ------     ----                   ----               -------
  Normal   Scheduled  5m58s                  default-scheduler  Successfully assigned confidential-containers-system/cc-operator-daemon-install-tr6fx to aks-nodepool1-10076017-vmss000000
  Normal   Pulled     5m20s                  kubelet            Successfully pulled image "quay.io/confidential-containers/runtime-payload-ci:kata-containers-8de1f8e19f858134ba455a7c04edcb21d8bcf6b1" in 31.094025085s (37.069779763s including waiting)
  Normal   Pulled     5m13s                  kubelet            Successfully pulled image "quay.io/confidential-containers/runtime-payload-ci:kata-containers-8de1f8e19f858134ba455a7c04edcb21d8bcf6b1" in 153.957989ms (153.964689ms including waiting)
  Normal   Pulled     5m1s                   kubelet            Successfully pulled image "quay.io/confidential-containers/runtime-payload-ci:kata-containers-8de1f8e19f858134ba455a7c04edcb21d8bcf6b1" in 123.53862ms (123.55092ms including waiting)
  Normal   Created    4m36s (x4 over 5m20s)  kubelet            Created container cc-runtime-install-pod
  Normal   Started    4m36s (x4 over 5m20s)  kubelet            Started container cc-runtime-install-pod
  Normal   Pulled     4m36s                  kubelet            Successfully pulled image "quay.io/confidential-containers/runtime-payload-ci:kata-containers-8de1f8e19f858134ba455a7c04edcb21d8bcf6b1" in 152.888065ms (152.894165ms including waiting)
  Normal   Pulling    3m44s (x5 over 5m57s)  kubelet            Pulling image "quay.io/confidential-containers/runtime-payload-ci:kata-containers-8de1f8e19f858134ba455a7c04edcb21d8bcf6b1"
  Normal   Pulled     3m44s                  kubelet            Successfully pulled image "quay.io/confidential-containers/runtime-payload-ci:kata-containers-8de1f8e19f858134ba455a7c04edcb21d8bcf6b1" in 167.848396ms (167.859396ms including waiting)
  Warning  BackOff    53s (x21 over 5m12s)   kubelet            Back-off restarting failed container cc-runtime-install-pod in pod cc-operator-daemon-install-tr6fx_confidential-containers-system(744a6ad7-0d5e-48c2-98b2-81d150c89213)
```
```
$ kubectl logs cc-operator-daemon-install-tr6fx -n confidential-containers-system
Environment variables passed to this script
* NODE_NAME: aks-nodepool1-10076017-vmss000000
* DEBUG: false
* SHIMS: remote
* DEFAULT_SHIM:
* CREATE_RUNTIMECLASSES: true
* CREATE_DEFAULT_RUNTIMECLASS: false
/opt/kata-artifacts/scripts/kata-deploy.sh: line 417: SNAPSHOTTER: unbound variable
```

**Comment by bpradipt** [comment]
Ah.. this is fixed in the main - https://github.com/confidential-containers/cloud-api-adaptor/blob/main/Makefile#L168

The version that you are using will most likely be using 
```
kubectl apply -k "github.com/confidential-containers/operator/config/release?def=v0.8.0"
```
instead
of 
```
kubectl apply -k "github.com/confidential-containers/operator/config/release?ref=v0.8.0"
```

**Comment by cmaf** [comment]
Thanks @bpradipt. I was able to run a pod with an older version of the instructions. Are there plans to either update the documentation or fix this in the next release?

**Comment by aep** [comment]
cloud-api-adaptor-0.8.0  is actually using `ref=v0.8.0`  and changing that to def doesnt fix it

where did you find the older instructions?

**Comment by bpradipt** [comment]
cc @stevenhorsman

**Comment by stevenhorsman** [comment]
Yeah, so the key fix to enable this (in https://github.com/confidential-containers/cloud-api-adaptor/commit/defb3f12e309c885e40126649fcee29a36876bd7) is `default` -> `release`, but the correct command that should resolve it is 
`kubectl apply -k "github.com/confidential-containers/operator/config/release?ref=v0.8.0"` as you said. 

Chelsea this is already fixed in CAA `main` and we're talking about doing interim 0.8.1 release to get it working in a 'stable' place, but need to line up a few dependencies first. As that point we can bump the version in the wider CoCo docs and it should work. In the short-term if you think it would help we could ask @surajssd to put a note in the 
https://confidentialcontainers.org/docs/cloud-api-adaptor/azure/#deploy-caa-on-the-kubernetes-cluster section to start that the `default`->`release` edit needs doing before deploying?

**Comment by surajssd** [comment]
- Created a PR that could fix this problem and will need updates to the documentation in the upstream: https://github.com/confidential-containers/cloud-api-adaptor/pull/1691

- But I have also updated the docs so that now user also has choice if they want to deploy latest stuff: https://github.com/confidential-containers/confidentialcontainers.org/pull/31

**Comment by cmaf** [comment]
@aep the chance to `kubectl apply -k "github.com/confidential-containers/operator/config/release?ref=v0.8.0"` did resolve it for me. The instructions I used a while back also worked, I pulled up this commit: https://github.com/confidential-containers/confidentialcontainers.org/commit/fd26c800dc569bf7c22ccc281847f2a965750c5b.

Thanks @stevenhorsman and @surajssd, I will take a look at the documentation PR.
